### DISCUSSION TITLE
Best exception to use to leave a xonsh script on error

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#4412] [Q&A] [answered]
Let's say I have a xonsh script which encounters an error (e.g. can't connect to database or whatever). I then want to leave the script with a short message and and stop it from continuing.

Is there an xonsh exception I can raise which does only print a given message but no the long stack-trace?

Thanks for any pointers!

**Comment by daniel-shimon** [comment]
So basically settings `$XONSH_SHOW_TRACEBACK=False` prevents xonsh from displaying stack traces

**Comment by jnoortheen** [comment, marked as answer]
sys.exit with 0 and a message will suffice

**Reply by halloleo** [reply]
Will try this. Thanks.

**Reply by halloleo** [reply]
Works well! Works for `xonsh script.xsh` _and_ for `source script.xsh`!

I guess this is rather the "shell-onic" way to do it, rather than the pythonic way... 😉
